### PR TITLE
Milestne: Refactor in module ES6

### PR DESCRIPTION
**Pull request by jicamargo**
# Refactor in modules ES6

## Refactor awesome books
This is a Refactor of the Awesome Books Application that includes:

- Use ES6 syntax.
- Divide code into modules.
- Sseparated files for each module in the modules directory.
- Import several modules into the index.js file. 
- Keep all funcionalities of the app without errors.
- Refactor functions to arrow functions.
- Use let and const in the correct way.
--------------

### CODE REVIEW

File: modules/addbook.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import saveBooks from './savebooks.js';
+
+export default function addBook(author, title, arrBooks) {
```
Comment (V-Blaze) [inline modules/addbook.js]: -  [x] Good job so far, :+1: :+1: Your entire project works great. :+1:  but I think there are some improvements you need to make on these functions according to the project requirements.
> Refactor your functions to arrow functions.

- - According to the project requirement, you are meant to refactor all your functions to use the ES6 arrow function, I'll suggest you follow the requirement and refactor these functions as arrow functions is a compact alternative to a traditional function expression, with some semantic differences:+1: :smile:

File: modules/getbooks.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function getBooks() {
```
Comment (V-Blaze) [inline modules/getbooks.js]: -  [x] As i suggested above, I think you should use the ES6 arrow function, I'll suggest you follow the requirement and refactor these functions as arrow functions is a compact alternative to a traditional function expression, with some semantic differences:+1: :smile: 
- Using arrow function here will look like this `const getBooks = () => { }`

File: modules/savebooks.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function saveBooks(arrBooks) {
```
Comment (V-Blaze) [inline modules/savebooks.js]: -  [x] As i suggested above, I think you should use the ES6 arrow function, I'll suggest you follow the requirement and refactor these functions as arrow functions is a compact alternative to a traditional function expression, with some semantic differences:+1: :smile:

File: modules/showbooklist.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import saveBooks from './savebooks.js';
+
+// builds the elements and show the books list inside the parent element
+
+export default function ShowBookList(parentElemId, arrBooks) {
```
Comment (V-Blaze) [inline modules/showbooklist.js:5]: -  [x] As i suggested above, I think you should use the ES6 arrow function, I'll suggest you follow the requirement and refactor these functions as arrow functions is a compact alternative to a traditional function expression, with some semantic differences:+1: :smile:

File: modules/updatedatetime.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// update the date and time using luxon
+
+import { DateTime } from './luxon.js';
+
+export default function updateDateTime() {
```
Comment (V-Blaze) [inline modules/updatedatetime.js]: -  [x] As i suggested above, I think you should use the ES6 arrow function, I'll suggest you follow the requirement and refactor these functions as arrow functions is a compact alternative to a traditional function expression, with some semantic differences:+1: :smile:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @jicamargo , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Project works fine after refactoring✔
- Functions are broken down into files in modules👋 
-  DateTime was setup with Luxon✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/addbook.js (hunk 1 above)
Comment (jicamargo) [reply modules/addbook.js]: thank you @V-Blaze . change done!

File: modules/getbooks.js (hunk 2 above)
Comment (jicamargo) [reply modules/getbooks.js]: thank you @V-Blaze . change done!

File: modules/savebooks.js (hunk 3 above)
Comment (jicamargo) [reply modules/savebooks.js]: thank you @V-Blaze . change done!

File: modules/showbooklist.js (hunk 4 above)
Comment (jicamargo) [reply modules/showbooklist.js:5]: thank you @V-Blaze . change done!

File: modules/updatedatetime.js (hunk 5 above)
Comment (jicamargo) [reply modules/updatedatetime.js]: thank you @V-Blaze  . change done!

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @jicamargo :wave: 
 
Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark: Descriptive pull request.
:heavy_check_mark: Correct Github flow.
:heavy_check_mark: The app is working fine.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**